txpool/txpool_test.go

```diff
@@ -1897,14 +1905,24 @@ func TestGetTxs(t *testing.T) {
 					}(tx)
 				}
 
-				totalTx += int(promotable)
+				totalPromotedTx += int(promotable)
 			}
 
 			ctx, cancelFn := context.WithTimeout(context.Background(), time.Second*10)
 			defer cancelFn()
 
-			// All txns should get added
-			assert.Len(t, waitForEvents(ctx, subscription, totalTx), totalTx)
+			// Wait for promoted transactions
+			assert.Len(t, waitForEvents(ctx, promoteSubscription, totalPromotedTx), totalPromotedTx)
+
+			// Wait for enqueued transactions, if any are present
+			totalEnqueuedTx := totalPromotedTx - len(test.allTxs)
```

Comment:
Nicpick, `totalPromotedTx` is just an expected value, right?

Reply:
Yes, it's the total number of transactions that should be promoted within the test case, so our `waitForEvents` call knows for how many events it should wait for

Reply:
The transaction can either be promoted or enqueued in this specific test, so there is no need to keep track of enqueued transactions when we already keep track of promoted ones - if a transaction is not promoted, then it is enqueued

Reply:
Fixed, in all places 👍 

[ccb1564](https://github.com/0xPolygon/polygon-edge/pull/421/commits/ccb1564169770421faa05b77fa9eb47abc4060a8)